From 2553ba15ee19ffd5b59ae72b7e0c50f76ef59e50 Mon Sep 17 00:00:00 2001
From: Jakob Kastelic <kastelic.jakob@gmail.com>
Date: Thu, 18 Sep 2025 21:08:05 -0700
Subject: [PATCH 1/1] select the boot partition in bootargs

Since we are booting without U-Boot, we cannot use it to select the boot
partition. One easy option is to put it in the kernel command line in
the DTS, as we do in this patch.

Signed-off-by: Jakob Kastelic <kastelic.jakob@gmail.com>
---
 arch/arm/boot/dts/st/stm32mp135f-dk.dts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/boot/dts/st/stm32mp135f-dk.dts b/arch/arm/boot/dts/st/stm32mp135f-dk.dts
index 1af335a39993..9c55120959e7 100644
--- a/arch/arm/boot/dts/st/stm32mp135f-dk.dts
+++ b/arch/arm/boot/dts/st/stm32mp135f-dk.dts
@@ -28,6 +28,7 @@ aliases {
 
 	chosen {
 		stdout-path = "serial0:115200n8";
+		bootargs = "root=/dev/mmcblk0p4 rootwait";
 	};
 
 	clocks {
-- 
2.39.5

